src/handler/patch.ts

```diff
@@ -0,0 +1,61 @@
+import 'dotenv/config';
+import { APIGatewayProxyEvent, APIGatewayProxyResult } from 'aws-lambda';
+import logger from '../util/logger';
+import { addHttpHeaders } from '../util/httpHeaders';
+import {
+  archiveOldCreateCurrentRecord,
+  getBySystemNumberAndCreatedTimestamp,
+} from '../services/database';
+import { validateUpdateVinRequest, validateVins } from '../validators/patch';
+import { processPatchVinRequest } from '../processors/processPatchVinRequest';
+import { TechrecordGet } from '../models/post';
+import { formatTechRecord } from '../util/formatTechRecord';
+
+export const handler = async (event: APIGatewayProxyEvent): Promise<APIGatewayProxyResult> => {
+  logger.info('Amend VIN Called');
+
+  const isRequestInvalid: APIGatewayProxyResult | undefined = validateUpdateVinRequest(event);
+
+  if (isRequestInvalid) {
+    return isRequestInvalid;
+  }
+
+  logger.debug('Request is Valid');
+
+  const systemNumber: string = decodeURIComponent(event.pathParameters?.systemNumber as string);
```

Comment:
I think it would be good to have some validation on these values before they get used, in order to ensure that they're not null/undefined/empty
```suggestion
  const systemNumber: string = event.pathParameters?.systemNumber ?? '';
  const createdTimestamp: string = event.pathParameters?.createdTimestamp ?? '';

  if (!systemNumber || !createdTimestamp) {
    return {
      statusCode: 400,
      body: 'some message',
    };
  }
```

Reply:
I did think about validating these exist, BUT to trigger the lambda you need to have a systemNumber and a createdTimestamp in the URL so is it necessary?

Reply:
I personally like checking it, also to make sure system number is correct. Maybe worth creating a little function that checks these, refactor the other end points that check, and then we just can call that?

Reply:
checked inside patch validator